state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -2838,15 +2824,43 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		}
 	}
 
+	fn refund_invalid_ccm(
+		ccm: CrossChainMessage<T::TargetChain, BlockNumberFor<T>>,
+		error: ExecutexSwapAndCallError,
+	) {
+		Self::deposit_event(Event::<T, I>::CcmEgressInvalid { egress_id: ccm.egress_id, error });
+		if let Ok(decoded_data) = T::CcmValidityChecker::decode_unchecked(
+			ccm.ccm_additional_data.clone(),
+			T::TargetChain::get(),
+		) {
+			if let Some(fallback_address) = decoded_data.refund_address::<T::TargetChain>() {
+				match Self::schedule_egress(
+					ccm.asset,
+					ccm.amount,
+					fallback_address.clone(),
+					None,
+					None
+				) {
+					Ok(egress_details) => Self::deposit_event(Event::<T, I>::InvalidCcmRefunded {
+						asset: ccm.asset,
+						amount: egress_details.egress_amount,
+						destination_address: fallback_address,
+					}),
+					Err(e) => log::warn!("Cannot refund failed Ccm: failed to Egress. aux_data_lookup_key: {:?}, Error: {:?}", ccm.aux_data_lookup_key, e),
+				};
+			}
+		} else {
+			log::warn!("Cannot refund failed Ccm: failed to decode `ccm_additional_data`. aux_data_lookup_key: {:?}", ccm.aux_data_lookup_key);
+		}
```

Comment:
This seems like a pretty bad case? How can this happen in reality?

Reply:
This should never happen. We guard against invalid CCM message in Vault swaps and when opening deposit channel. This is just there just in case.

Reply:
> We guard against invalid CCM message in Vault swaps and when opening deposit channel

Where is that code?

Reply:
Yeah the check is done with the `CcmValidityChecker::check_and_decode` function. It's done at the following places:
`process_vault_swap_request_prewitness()` (IngressEgress::vault_swap_request)
`process_vault_swap_request_full_witness()` (IngressEgress::vault_swap_request)
`pallet_cf_swapping::request_swap_deposit_address_with_affiliates()`
`cf_request_swap_parameter_encoding()` in our RuntimeApi for requesting Vault Swap encodings